### PR TITLE
Fix vcs urls parsing with extras_require

### PR DESCRIPTION
**Pull request by ticosax**
Make sure extras are correctly parsed in vcs like urls
`git+https://git.myproject.org/MyProject.git@v0.1#egg=MyProject[security]`

On top of #29 and #28 .This pull request needs to be rebased, once they are merged (I hope so).

### CODE REVIEW

File: requirements/fragment.py [hunk 1]
```diff
@@ -27,3 +27,10 @@ def get_hash_info(d):
             return key, d[key]
 
     return None, None
+
+
+def parse_extras_require(egg):
+    if egg is not None and ('[' and ']') in egg:
```
Comment (davidfischer) [inline requirements/fragment.py]: This line is not quite right:

    ('[' and ']') in 'eggblah]'  # True

Possibly just use a regex?

Comment (ticosax) [inline requirements/fragment.py]: yes, it seems more appropriate.

**Comment by ticosax** [comment]
Gentle ping here, to check if you noticed I updated the PR.
Have a good day.

**Comment by davidfischer** [comment]
Thanks for the ping. I'll try to take a look in the next day or two.

**Comment by kennethreitz** [comment]
Let's get this merged!

**Comment by davidfischer** [comment]
This looks good. Thanks for making the changes.

My merging is only two months late. 💥

**Pull request merged**